server/entitymanager/entitymanager.go

```diff
@@ -102,43 +133,63 @@ func populateBootConfig(conf *epb.BootConfig) (*bootz.BootConfig, error) {
 }
 
 // GetBootstrapData fetches and returns the bootstrap data response from the server.
-func (m *InMemoryEntityManager) GetBootstrapData(chassis *service.EntityLookup, controllerCard *bootz.ControlCard) (*bootz.BootstrapDataResponse, error) {
+func (m *InMemoryEntityManager) GetBootstrapData(lo *service.EntityLookup, controllerCard *bootz.ControlCard) (*bootz.BootstrapDataResponse, error) {
 	// First check if we are expecting this control card.
-	if controllerCard.SerialNumber == "" {
-		return nil, status.Errorf(codes.InvalidArgument, "no serial number provided")
+	serial := ""
+	fixedChasis := false
+	if controllerCard == nil {
+		if lo.SerialNumber == "" {
+			return nil, status.Errorf(codes.NotFound, "chassis type (fixed/modular) can not be determined, either controller card or chassis serial must be set ")
```

Comment:
Probably codes.invalidargument is more applicable here

Reply:
sure